core/src/main/scala/commandcenter/command/CalculatorCommand.scala

```diff
@@ -1,29 +1,129 @@
+      parsed                = decline.Command(title, title)(helpTypeOpt).parse(input.args)
+      (helpTitle, message) <- ZIO
+                                .fromEither(parsed)
+                                .fold(
+                                  help => (title, HelpMessage.formatted(help)),
+                                  helpType => (helpType.title, CalculatorCommand.helpMessageByType(helpType))
+                                )
+    } yield List(
+      Preview(BigDecimal(0.0)).score(Scores.high(searchInput.context)).view(DefaultView(helpTitle, message))
+    )
+
+  private def previewEvaluation(searchInput: SearchInput): ZIO[Env, CommandError, List[PreviewResult[BigDecimal]]] =
     for {
-      evaluatedValue <-
-        ZIO.fromOption(CalculatorUtil.evaluate(searchInput.input)).orElseFail(CommandError.NotApplicable)
+      evaluatedValue <- ZIO
+                          .fromOption(new CalculatorUtil(parameters).evaluate(searchInput.input))
+                          .orElseFail(CommandError.NotApplicable)
     } yield List(
       Preview(evaluatedValue)
-        .onRun(tools.setClipboard(evaluatedValue.toString))
+        .render(parameters.format)
+        .onRun(tools.setClipboard(parameters.format(evaluatedValue)))
         .score(Scores.high(searchInput.context))
     )
 }
 
 object CalculatorCommand extends CommandPlugin[CalculatorCommand] {
-  def make(config: Config): TaskManaged[CalculatorCommand] = ZManaged.succeed(CalculatorCommand())
+  def make(config: Config): TaskManaged[CalculatorCommand] = ZManaged.fromEither(config.as[CalculatorCommand])
+
+  implicit val decoder: Decoder[CalculatorCommand] = Decoder.instance { c =>
+    val decimalFormat        = new DecimalFormat()
+    val decimalFormatSymbols = decimalFormat.getDecimalFormatSymbols
+    for {
+      decimalSeparator      <- c.get[Option[Char]]("decimalSeparator")
+      groupingSeparator     <- c.get[Option[Char]]("groupingSeparator")
+      parameterSeparator    <- c.get[Option[Char]]("parameterSeparator")
+      groupingSize          <- c.get[Option[Int]]("groupingSize")
+      groupingUsed          <- c.get[Option[Boolean]]("groupingUsed")
+      maximumFractionDigits <- c.get[Option[Int]]("maximumFractionDigits")
+    } yield CalculatorCommand(
+      Parameters(
+        decimalSeparator.getOrElse(decimalFormatSymbols.getDecimalSeparator),
+        groupingSeparator.getOrElse(decimalFormatSymbols.getGroupingSeparator),
+        parameterSeparator.getOrElse(';'),
+        groupingSize.getOrElse(decimalFormat.getGroupingSize),
+        groupingUsed.getOrElse(decimalFormat.isGroupingUsed),
+        maximumFractionDigits.getOrElse(decimalFormat.getMaximumFractionDigits)
+      )
+    )
+  }
+
+  def helpMessageByType(helpType: HelpType): fansi.Str =
+    helpType match {
+      case FunctionsList  => CalculatorUtil.helpMessageFunctionsList
+      case ParametersList => CalculatorUtil.helpMessageParametersList
+    }
+}
+
+final case class Parameters(
```

Comment:
I would put stuff like `Parameters` and `HelpType` inside `object CalculatorCommand` scope. Otherwise if some other command tried to name something in the same way there would be a collision.

Or if you want to put it in a separate file, creating a `commandcenter.command.calculator` package would work too. But so far every command has been a single file.

Reply:
Thanks, @reibitto. Yes, a common syntax for help would be nice. Ok, you're right, I'll move these classes inside the object.